board/gpio.md

```diff
@@ -3,70 +3,97 @@ GPIO API
 
-<a name="gpio"></a>
-### The `GPIO` interface
-Represents the properties and methods that expose GPIO functionality. The `GPIO` object implements the [`EventEmitter`](../README.md/#events) interface, and extends the [`Pin`](./README.md/#pin) object.
+<a name="apiobject"></a>
+### The GPIO API object
+When requiring `"gpio"`, the following steps are run:
+- If there is no permission for using the functionality, throw `SecurityError`.
+- If the GPIO functionality is not supported on the board, throw `"NotSupportedError"`.
+- Return an object that implements the following methods.
+
+| Method              | Description      |
+| ---                 | ---              |
+| [`open()`](#open)   | open GPIO pin    |
+| [`port()`](#port)   | open GPIO port   |
+
+See also the [Web IDL](./webidl.md) definition.
+
+The following dictionary is used for initializing GPIO pins and ports.
+<a name="gpiooptions"></a>
 
 | Property   | Type   | Optional | Default value | Represents |
 | ---        | ---    | ---      | ---           | ---        |
-| `pin`      | String or Number | no | `undefined`   | board name for the pin |
+| `pin `     | String or Number | no | `undefined`   | pin name |
+| `mapping`  | String | no | `"system"`   | pin mapping |
 | `mode`     | String | no       | `undefined`   | I/O mode |
-| `port`     | array  | yes      | `undefined`   | array of pin names representing the ports
 | `activeLow` | boolean | yes   | `false` | whether the pin is active on logical low |
 | `edge`     | string | yes      | `"any"`       | Interrupt generation mode |
 | `state`    | string | yes      | `undefined`      | "pulldown", "pullup" |
 
-| Method signature     | Description       |
-| ---                  | ---               |
-| [`read()`](#read)    | synchronous read  |
-| [`write()`](#write)  | synchronous write |
-| [`close()`](#close)  | close the pin     |
+The `pin` property is either a number or string, with values defined by the OS or board documentation. The default value is `undefined`.
 
-| Event name | Event callback argument |
-| -----------| ----------------------- |
-| `data`     | unsigned long (the pin value) |
+The `mapping` property represents the pin namespace, either `"system"` or `"board"`, by default `"system"`.
 
-The `data` event listener callback receives the current value of the pin (0 or 1 for single pins, and positive integer for GPIO ports).
+The `mode` property MUST take the value `"in"` or `"out"`. The default value is `"out"`.
 
-The `pin` property inherited from [`Pin`](./README.md/#pin) can take values defined by the board documentation, usually positive integers.
+The `activeLow` property tells whether the pin value 0 means active. If `activeLow` is `true`, with `value` 0 the pin is active, otherwise inactive. For instance, if an actuator is attached to the (output) pin active on low, client code should write the value 0 to the pin in order to activate the actuator. The default value is `false`.
```

Comment:
But we could still use `activeLow` for configuring a pin, where possible. And we could keep it as a read-only property.

The question is whether is there a use case for the set/reset API, and you've just given one. Should I include them?